cppapi/client/zmqeventconsumer.cpp

```diff
@@ -698,6 +698,9 @@ bool ZmqEventConsumer::process_ctrl(zmq::message_t &received_ctrl,zmq::pollitem_
 {
     bool ret = false;
 
+    // static variable to count the number of ZMQ_DELAY_EVENT requests currently in progress:
+    static int nb_current_delay_event_requests = 0;
```

Comment:
Can you change this to a member variable? This was discussed already during the review of the original PR but postponed due to ABI compatibility. Adding more global state is going to be a problem if we at some point decide to delete or recreate instance of this class.

Reply:
@bourtemb Although I created the PR, can I pass that onto you?

Reply:
Sure I'll do that.